include/boost/json/detail/impl/number.ipp

```diff
@@ -230,7 +237,34 @@ loop:
         goto finish;
     }
 
-    //-----------------------------------
+    // 1*digit
+    case state::mantf:
```

Comment:
if you really want to make up for the comment removed from the header, you could write
```
// 1*digit
// A digit after the decimal point is required.
case state::mantf
```
Although it isn't strictly needed since the BNF already explains it.